packages/host/app/components/operator-mode/code-submode.gts

```diff
@@ -320,6 +325,23 @@ export default class CodeSubmode extends Component<Signature> {
     this.userHasDismissedURLError = true;
   }
 
+  @action private onModuleEdit(state: State) {
+    let editedDeclaration = state.declarations.find(
+      (newDeclaration: ModuleDeclaration) => {
+        return this.selectedDeclaration
+          ? this.selectedDeclaration.localName !== newDeclaration.localName &&
+              isEquivalentBody(
+                this.selectedDeclaration.path,
+                newDeclaration.path,
+              )
+          : false;
+      },
+    );
+    if (editedDeclaration) {
+      this.goToDefinition(undefined, editedDeclaration.localName);
+    }
+  }
```

Comment:
This is the main hook that gets activated when u edit a module. It checks if you  have changed the name of a declaration but the body still remains.